pkg/cli/tail.go

```diff
@@ -222,24 +275,46 @@ func (c *TailCommand) Run(ctx context.Context, args []string) error {
 	return nil
 }
 
-func (c *TailCommand) tail(ctx context.Context) ([]*loggingpb.LogEntry, error) {
-	var p logPuller
-	if c.testPuller != nil {
-		p = c.testPuller
-	} else {
-		logClient, err := logging.NewClient(ctx)
+func (c *TailCommand) stream(ctx context.Context, extra []validation.Validator, puller logPuller) error {
+	// logCh is used to receive logEntriesfrom StreamPull
+	logCh := make(chan *loggingpb.LogEntry)
+
+	var failCount, totalCount int
+
+	var perr error
+	go func() {
+		if err := puller.StreamPull(ctx, c.queryFilter(), logCh); err != nil {
+			perr = fmt.Errorf("StreamPull failed: %w", err)
+		}
+		close(logCh)
+	}()
+
+	for l := range logCh {
+		totalCount++
```

Comment:
@sqin2019 Is it intentional we're ignoring max-num here?

Reply:
Should we not? I was thinking a different behavior as mentioned in [here](https://github.com/abcxyz/lumberjack/pull/448/files#r1309449205). Do you recommend to use max-num for both tail and streamtail?

Reply:
ah... Maybe we should consider maxnum here since linux tail does.

Reply:
I think it's confusing that you can still specify max-num even with -f flag.